tests/test_mirror.py

```diff
@@ -6,228 +6,762 @@
+        assert self.get_mirror_session_status(session) == "inactive"
+
+        # bring up port; add ip; add neighbor; add route
+        self.set_interface_status("Ethernet32", "up")
+        self.add_ip_address("Ethernet32", "80.0.0.0/31")
+        self.add_neighbor("Ethernet32", "80.0.0.1", "02:04:06:08:10:12")
+        self.add_route(dvs, "8.8.0.0/16", "80.0.0.1")
+        assert self.get_mirror_session_status(session) == "active"
+
+        # check monitor port
+        tbl = swsscommon.Table(self.adb, "ASIC_STATE:SAI_OBJECT_TYPE_MIRROR_SESSION")
+        assert len(tbl.getKeys()) == 1
+        (status, fvs) = tbl.get(tbl.getKeys()[0])
+        assert status == True
+        for fv in fvs:
+            if fv[0] == "SAI_MIRROR_SESSION_ATTR_MONITOR_PORT":
+                assert dvs.asicdb.portoidmap[fv[1]] == "Ethernet32"
+
+        # mirror session move round 1
+        # create vlan; create vlan member; bring up vlan and member
+        self.create_vlan(dvs, "9")
+        self.create_vlan_member("9", "Ethernet48")
+        self.set_interface_status("Vlan9", "up")
+        self.set_interface_status("Ethernet48", "up")
+        assert self.get_mirror_session_status(session) == "active"
+
+        # add ip address to vlan 9
+        self.add_ip_address("Vlan9", "8.8.8.0/24")
+        assert self.get_mirror_session_status(session) == "inactive"
+
+        # create neighbor to vlan 9
+        self.add_neighbor("Vlan9", "8.8.8.8", "88:88:88:88:88:88")
+        assert self.get_mirror_session_status(session) == "inactive"
+
+        # create fdb entry to ethernet48
+        self.create_fdb("9", "88-88-88-88-88-88", "Ethernet48")
+        assert self.get_mirror_session_status(session) == "active"
+
+        # check monitor port
+        tbl = swsscommon.Table(self.adb, "ASIC_STATE:SAI_OBJECT_TYPE_MIRROR_SESSION")
+        assert len(tbl.getKeys()) == 1
+        (status, fvs) = tbl.get(tbl.getKeys()[0])
+        assert status == True
+        for fv in fvs:
+            if fv[0] == "SAI_MIRROR_SESSION_ATTR_MONITOR_PORT":
+                assert dvs.asicdb.portoidmap[fv[1]] == "Ethernet48"
+
+        # mirror session move round 2
+        # remove fdb entry
+        self.remove_fdb("9", "88-88-88-88-88-88")
+        assert self.get_mirror_session_status(session) == "inactive"
+
+        # remove neighbor
+        self.remove_neighbor("Vlan9", "8.8.8.8")
+        assert self.get_mirror_session_status(session) == "inactive"
+
+        # remove ip address
+        self.remove_ip_address("Vlan9", "8.8.8.0/24")
+        assert self.get_mirror_session_status(session) == "active"
+
+        # check monitor port
+        tbl = swsscommon.Table(self.adb, "ASIC_STATE:SAI_OBJECT_TYPE_MIRROR_SESSION")
+        assert len(tbl.getKeys()) == 1
+        (status, fvs) = tbl.get(tbl.getKeys()[0])
+        assert status == True
+        for fv in fvs:
+            if fv[0] == "SAI_MIRROR_SESSION_ATTR_MONITOR_PORT":
```

Comment:
can you assert SAI_MIRROR_SESSION_ATTR_VLAN_HEADER_VALID == false here?

Reply:
sure i'll add this